src/main/webapp/index_script.js

```diff
@@ -13,37 +13,29 @@
 // limitations under the License.
 
 document.addEventListener("DOMContentLoaded", async function() {
-  let organizations = [{
-    "name": "Organization A",
-  },
-  {
-    "name": "Organization B",
-  },
-  {
-    "name": "Organization C",
-  },
-  {
-    "name": "Organization D",
-  },
-  {
-    "name": "Organization E",
-  },
-  {
-    "name": "Organization F"
-  }];
   const organizationArea = document.getElementById('organization-list');
-  organizations.forEach((organization) => {
-    organizationArea.appendChild(createOrganization(organization));
-  });
+  renderOrganizations(organizationArea);
 });
 
+/* 
+ * This async function gets a default list of organization names to display when the dom content
+ * loads by not passing any parameters to /list-organizations
+ */
+async function renderOrganizations(organizationArea) {
+  const response = await fetch(`/list-organizations`);
+  const organizationNames = await response.json();
+  for (let i = 0; i < organizationNames.length; i++) {
+    organizationArea.appendChild(createOrganization(organizationNames[i]));
+  }
+}
+
 function createOrganization(organization) {
   const organizationElement = document.createElement("div");
   organizationElement.classList.add("organization");
 
   const organizationNameElement = document.createElement('div');
   organizationNameElement.classList.add("organization-name");
-  organizationNameElement.textContent = organization.name;
+  organizationNameElement.textContent = organization;
```

Comment:
it would have to be organization.name because organization is an object

Reply:
Since ListOrganizationServlet just sends back an array of strings that each are an organization name, this function is not dealing with an object- "organization" is just one of the strings in the array.